### DISCUSSION TITLE
`Mapper` entity type is `DeclarativeAttributeIntercept` on `do_orm_execute`

### DISCUSSION THREAD

**Discussion by jellevwezel** [sqlalchemy/sqlalchemy#10129] [Usage Questions] [answered]
Hi,

I'm trying to use the `'do_orm_execute'` event on a session, and I'm observing some unexpected (to me) behavior.
When I try to access the `entity` property through a `Mapper` obtained from the `ORMExecuteState` the type of the entity is `DeclarativeAttributeIntercept` where I expect it to be the actual `entity`

Versions:
```
sqlalchemy: 2.0.19
python: 3.11
os: ubuntu 20.04
```

Here is my example code:

```python
from sqlalchemy import create_engine, event
from sqlalchemy.orm import mapped_column, DeclarativeBase, Mapped, ORMExecuteState, Session


class Base(DeclarativeBase):
    pass


class Foo(Base):
    __tablename__ = 'foo'
    id: Mapped[int] = mapped_column(primary_key=True)


def do_orm_execute_event(state: ORMExecuteState):
    mapper = state.all_mappers[0]
    entity = mapper.entity
    print(type(entity))  # expect to print 'Foo', but prints: 'DeclarativeAttributeIntercept'


engine = create_engine('sqlite://')
Base.metadata.create_all(engine)

event.listen(Session, 'do_orm_execute', do_orm_execute_event)

with Session(engine) as session:
    session.add(Foo(id=1))
    session.commit()  # event not triggered on the insert?
    foo = session.get(Foo, 1)
```

However, my debugger does seem to know it is a `Foo`:
![image](https://github.com/sqlalchemy/sqlalchemy/assets/70206079/debee9c6-e4d3-4222-9d3f-516b2793bc15)

Am I doing something wrong, or am I not supposed to use the `Mapper` properties in this way?
Please let me know :)

**Comment by CaselIT** [comment, marked as answer]
Hi,

This the expected behaviour, since entity is a class, so doing type returns its metaclass
As documented mapper.entity is an alias for mapper.class_ https://docs.sqlalchemy.org/en/20/orm/mapping_api.html#sqlalchemy.orm.Mapper.entity

**Comment by jellevwezel** [comment]
Thanks, I was just a bit thrown off by the `DeclarativeAttributeIntercept` metaclass, and thinking one level to shallow :)
